gpytorch/mlls/exact_marginal_log_likelihood.py

```diff
@@ -39,8 +39,12 @@ def _add_other_terms(self, res, params):
             res = res.add(added_loss_term.loss(*params))
 
         # Add log probs of priors on the (functions of) parameters
+        res_ndim = res.ndim
         for name, module, prior, closure, _ in self.named_priors():
-            res.add_(prior.log_prob(closure(module)).sum())
+            prior_term = prior.log_prob(closure(module))
```

Comment:
How about 
```
prior_term = prior.log_prob(closure(module))
res.add_(prior_term.view(*res.ndim, -1).sum(dim=-1))
```

Reply:
Oops, forgot about this for a while. Updated!